src/HttpCache/EventListener/AddTagsListener.php

```diff
@@ -25,16 +27,21 @@
  * @see https://support.cloudflare.com/hc/en-us/articles/206596608-How-to-Purge-Cache-Using-Cache-Tags-Enterprise-only-
  *
  * @author Kévin Dunglas <dunglas@gmail.com>
+ * @author Yanick Witschi <yanick.witschi@terminal42.ch>
  *
  * @experimental
  */
 final class AddTagsListener
 {
     private $iriConverter;
+    private $purger;
+    private $debug;
 
-    public function __construct(IriConverterInterface $iriConverter)
+    public function __construct(IriConverterInterface $iriConverter, PurgerInterface $purger, bool $debug = false)
```

Comment:
Can't we just typehint `CacheTagsFormattingPurgerInterface` directly?

Reply:
No because the current `VarnishPurger` does not implement that and I don't know what other implementations have been made out there. Also it's completely optional to adjust the formatting of the tags :)

Reply:
I don't get it. The dependency wasn't existent before. `CacheTagsFormatterInterface $formatter = null` will do the trick. Btw we try to avoid setter injection as much as possible.